### PR TITLE
[Animation] Expose `SkeletonProfile.set/is_required`

### LINKED ISSUE #89892: SkeletonProfile's is_require(), and set_require() aren't exposed
### Tested versions

4.2.1 Stable

### System information

Ubuntu Linux, Mobile+

### Issue description

While setting parameters for a Bone Map's SkeletonProfile in code, i found the set_require(), and is_require() don't seem to be available. The Docs don't list them as available , but the resource itself does seem to have it coded for use, (but i don't know C#/C++, so i could just be a dumb-dumb) but  those fields are handy for setting values in code rather than manually in the inspector per each 50+ bones of a skeleton profile. I don't understand well enough why they're not usable:

https://github.com/godotengine/godot/blob/7c6e85de0dd7627d1a43acf5f144cee34df6ab36/scene/resources/skeleton_profile.cpp#L373

What i'd like to see as far as expected behavior goes, is that the "require" checkbox for the bone in the inspector, be set-able in code. Either a set_require, or other method to directly set the value outside of the GUI. Or for somebody to explain how to do it since i could just be missing something.

### Steps to reproduce

Create a script, export a skeleton profile, and then begin setting parameters for the bones inside it in code. You can set anything they ask for in the Inspector except for the 'Require" checkbox. In my case making a tool to do this for me, i used this code. My MRP will contain this in action and circumstance.

```
@tool
extends Skeleton3D

@export var refresh :bool = true : set = set_refresh # my cheap refresh button
@export var bone_template  = SkeletonProfile.new()

func _set_test(new):
	print()

func set_refresh(_new):
	if Engine.is_editor_hint():
		bone_template.bone_size = get_bone_count()

		var count = get_bone_count()
		for bone in count:
			bone_template.set_bone_name(bone,get_bone_name(bone))
			bone_template.set_bone_parent(bone,get_bone_name(get_bone_parent(bone)))
			bone_template.set_reference_pose(bone,get_bone_global_pose(bone))
			bone_template.set_handle_offset(bone,Vector2(.5,.1 +(bone *  .1)))
			# THe following command or something like it:
			#bone_template.set_require(bone,true)
```

### Minimal reproduction project (MRP)

[SkelProfile.zip](https://github.com/godotengine/godot/files/14750887/SkelProfile.zip)

### PR DESCRIPTION
**Pull request by AThousandShips**
* Fixes: https://github.com/godotengine/godot/issues/89892

### CODE REVIEW

File: doc/classes/SkeletonProfile.xml [hunk 1]
```diff
@@ -83,6 +83,13 @@
 				Returns the texture of the group at [param group_idx] that will be the drawing group background image in the [BoneMap] editor.
 			</description>
 		</method>
+		<method name="is_require" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="bone_idx" type="int" />
+			<description>
+				Returns whether the bone at [param bone_idx] is required.
```
Comment (AThousandShips) [inline doc/classes/SkeletonProfile.xml:90]: Unsure about how exactly to phrase this

File: scene/resources/skeleton_profile.cpp [hunk 2]
```diff
@@ -432,6 +432,9 @@ void SkeletonProfile::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_group", "bone_idx"), &SkeletonProfile::get_group);
 	ClassDB::bind_method(D_METHOD("set_group", "bone_idx", "group"), &SkeletonProfile::set_group);
 
+	ClassDB::bind_method(D_METHOD("is_require", "bone_idx"), &SkeletonProfile::is_require);
+	ClassDB::bind_method(D_METHOD("set_require", "bone_idx", "require"), &SkeletonProfile::set_require);
```
Comment (AThousandShips) [inline scene/resources/skeleton_profile.cpp:436]: Should we follow the naming if the c++ methods here or use `is_required`?

File: scene/resources/skeleton_profile.h
Comment (AThousandShips) [inline scene/resources/skeleton_profile.h:1]: Went ahead and cleaned up some arguments while I was at it here

File: scene/resources/skeleton_profile.cpp (hunk 2 above)
Comment (akien-mga) [reply scene/resources/skeleton_profile.cpp:436]: It should be `set`/`is_required`.
I would also rename the C++ methods and struct member, if we don't mind a minor compat breakage for C++ modules.

Comment (AThousandShips) [reply scene/resources/skeleton_profile.cpp:436]: Will do!

File: doc/classes/SkeletonProfile.xml [hunk 3]
```diff
@@ -83,6 +83,13 @@
 				Returns the texture of the group at [param group_idx] that will be the drawing group background image in the [BoneMap] editor.
 			</description>
 		</method>
+		<method name="is_required" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="bone_idx" type="int" />
+			<description>
+				Returns whether the bone at [param bone_idx] is required.
```
Comment (TokageItLab) [inline doc/classes/SkeletonProfile.xml:90]:
```suggestion
				Returns whether the bone at [param bone_idx] is required for retargeting.
```

**Review by TokageItLab — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**